src/plugins/ramps/simplex/simplexRampPlugin.ts

```diff
@@ -0,0 +1,791 @@
+            // For checkSupport, we just need to know if the plugin is supported
+            // The actual currency code mapping happens during quote
+            displayCurrencyCode = Object.keys(pluginMappings)[0]
+          }
+        } else {
+          // Simplex doesn't support tokens, only native currencies
+          return { supported: false }
+        }
+
+        if (!displayCurrencyCode) {
+          return { supported: false }
+        }
+
+        // Validate crypto - we use any valid display code for the plugin
+        const simplexCryptoCode = validateCrypto(
+          cryptoAsset.pluginId,
+          displayCurrencyCode
+        )
+        if (!simplexCryptoCode) {
+          return { supported: false }
+        }
+
+        // Validate fiat - ensure 'iso:' prefix
+        const simplexFiatCode = validateFiat(
+          ensureIsoPrefix(fiatAsset.currencyCode)
+        )
+        if (!simplexFiatCode) {
+          return { supported: false }
+        }
+
+        // All validations passed
+        return { supported: true }
+      } catch (error) {
+        // Only throw for actual errors (network issues, etc)
+        // Never throw for unsupported combinations
+        console.error('Simplex checkSupport error:', error)
+        throw error
+      }
+    },
+
+    fetchQuote: async (
+      request: RampQuoteRequest
+    ): Promise<RampQuoteResult[]> => {
+      const {
+        amountType,
+        exchangeAmount,
+        regionCode,
+        pluginId: currencyPluginId,
+        fiatCurrencyCode,
+        displayCurrencyCode,
+        direction
+      } = request
+
+      const isMaxAmount =
+        typeof exchangeAmount === 'object' && exchangeAmount.max
+      const exchangeAmountString = isMaxAmount ? '' : (exchangeAmount as string)
+
+      // Validate direction
+      if (!validateDirection(direction)) {
+        return []
+      }
+
+      // Initialize state and fetch provider config if needed
+      await ensureStateInitialized()
+      await fetchProviderConfig()
+
+      // Ensure we have provider config
+      if (!providerConfig) {
+        console.error('Simplex: Provider config not available')
+        return []
+      }
+
+      // Validate region
+      if (!validateRegion(regionCode)) {
+        return []
+      }
+
+      // Validate crypto
+      const simplexCryptoCode = validateCrypto(
+        currencyPluginId,
+        displayCurrencyCode
+      )
+      if (!simplexCryptoCode) {
+        return []
+      }
+
+      // Validate fiat - ensure 'iso:' prefix
+      const simplexFiatCode = validateFiat(ensureIsoPrefix(fiatCurrencyCode))
+      if (!simplexFiatCode) {
+        return []
+      }
+
+      // All checks passed, now fetch the actual quote
+      if (!state) throw new Error('Plugin state not initialized')
+
+      // Prepare quote request
+      const ts = Math.floor(Date.now() / 1000)
+      let socn: string, tacn: string
+      let soam: number
```

Comment:
Improve these var names.

Also, one decl per line.

Reply:
Yeah f*ck that noise.